### PR TITLE
Adding System.DirectoryServices.Protocols Tests

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @AlexGhiondea @danmosemsft @hughbe @WinCPP

**Comment by tarekgh** [comment]
#24590
#24316
#24388
#24533
#24692
#24762 
#20669

**Comment by tarekgh** [comment]
@tquerec ping!

**Comment by tarekgh** [comment]
@tquerec please have a look if you have a few minutes. thanks.

File: src/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+                try
+                {
+                    AddOrganizationalUnit(connection, dn);
+                    SearchResultEntry sre = SearchOrganizationalUnit(connection, LdapConfiguration.Configuration.Domain, ouName);
+                    Assert.NotNull(sre);
+
+                    AddOrganizationalRole(connection, user1Dn);
+                    AddOrganizationalRole(connection, user2Dn);
+
+                    string usersRoot = dn + "," + LdapConfiguration.Configuration.Domain;
```
Comment (tquerec) [inline src/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs]: We should setup an LDS instance and make sure this works.  LDS has some specific requirements on how paths are used.

Comment (tarekgh) [reply src/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs]: could you please explain more how I can do that? or if you have a pointer to some example that will be great

**Comment by tarekgh** [comment]
I am merging this change and will look at the LDS issue separately. Thanks @tquerec for your review

File: src/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs]: do you mean here connecting to the domain which the local machine is connected to?

Comment (tarekgh) [reply src/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs]: Is what you mean here is doing https://technet.microsoft.com/en-us/library/cc725619(v=ws.10).aspx?

**Pull request merged**